src/test/java/com/exasol/adapter/dialects/exasol/AbstractExasolSqlDialectIntegrationTest.java

```diff
@@ -64,61 +54,61 @@
     private static final String SCHEMA_TEST_MIXED_CASE = "SCHEMA_TEST_Mixed_Case";
     private static final String TABLE_MIXED_CASE = "Table_Mixed_Case";
     private static final String VIRTUAL_SCHEMA_EXA_MIXED_CASE = "VIRTUAL_SCHEMA_EXA_Mixed_Case";
+    private static final String EXASOL_DIALECT = "EXASOL";
 
-    public static final String DEBUGGER_PORT = "8000";
-    public static final String EXASOL_DIALECT = "EXASOL";
-    @Container
-    private static final ExasolContainer<? extends ExasolContainer<?>> container = new ExasolContainer<>(
-            ExasolContainerConstants.EXASOL_DOCKER_IMAGE_REFERENCE).withLogConsumer(new Slf4jLogConsumer(LOGGER))
-                    .withReuse(true);
     private static Statement statement;
     private static Connection connection;
     private static ExasolObjectFactory exasolObjectFactory;
     private static AdapterScript adapterScript;
     private static ConnectionDefinition connectionDefinition;
+    private static ExasolContainer<? extends ExasolContainer<?>> container;
 
-    private static String getTestHostIp() {
-        final Map<String, ContainerNetwork> networks = container.getContainerInfo().getNetworkSettings().getNetworks();
-        if (networks.size() == 0) {
-            return null;
-        }
-        return networks.values().iterator().next().getGateway();
-    }
+    // Set in the implementations classes
+    protected static String hostAndDefaultPort;
+    protected static String dockerImageReference;
 
-    @BeforeAll
-    static void beforeAll()
-            throws SQLException, BucketAccessException, InterruptedException, TimeoutException, IOException {
+    protected static void startSetUp()
+            throws SQLException, BucketAccessException, InterruptedException, TimeoutException {
+        container = new ExasolContainer<>(
+                dockerImageReference).withLogConsumer(new Slf4jLogConsumer(LOGGER)).withReuse(true);
```

Comment:
If your are testing with different variants of the container `withReuse()` makes no sense.